src/components/__tests__/SortBy.js

```diff
@@ -0,0 +1,92 @@
+import { mount } from '@vue/test-utils';
+import SortBy from '../SortBy.vue';
+import { __setState } from '../../component';
+
+jest.mock('../../component');
+
+const defaultState = {
+  options: [
+    { value: 'some_index', label: 'Relevance' },
+    { value: 'some_index_cool', label: 'Coolness ascending' },
+    { value: 'some_index_quality', label: 'Quality ascending' },
+  ],
+  hasNoResults: false,
+  currentRefinement: 'some_index',
+};
+
+const defaultProps = {
+  items: [
+    { value: 'some_index', label: 'Relevance' },
+    { value: 'some_index_cool', label: 'Coolness ascending' },
+    { value: 'some_index_quality', label: 'Quality ascending' },
+  ],
+};
+
+it('renders correctly', () => {
+  __setState({ ...defaultState });
+
+  const wrapper = mount(SortBy, {
+    propsData: {
+      items: [
+        { value: 'some_index', label: 'Relevance' },
+        { value: 'some_index_cool', label: 'Coolness ascending' },
+        { value: 'some_index_quality', label: 'Quality ascending' },
+      ],
+    }, // ok
+  });
+  expect(wrapper.html()).toMatchSnapshot();
+});
+
+it('renders with scoped slots', () => {
+  const defaultSlot = `
+  <select
+    slot-scope="{ items, refine, currentRefinement }"
+    @change="refine($event.currentTarget.value)"
+  >
+    <option
+      v-for="item in items"
+      :key="item.value"
+      :value="item.value"
+      :selected="item.value === currentRefinement"
+    >
+      {{item.label}}
+    </option>
+  </select>
+`;
+
+  __setState({
+    ...defaultState,
+  });
+
+  const wrapper = mount(SortBy, {
+    propsData: {
+      ...defaultProps,
+    },
+    scopedSlots: {
+      default: defaultSlot,
+    },
+  });
+
+  expect(wrapper.html()).toMatchSnapshot();
+});
+
+it('behaves correctly', () => {
+  const refine = jest.fn();
+  __setState({
+    ...defaultState,
+    refine,
+  });
+  const wrapper = mount(SortBy, {
+    propsData: {
+      ...defaultProps,
+    },
+  });
+  // This is bad 👇🏽 but the only way for now to trigger changes
+  // on a select: https://github.com/vuejs/vue-test-utils/issues/260
+  const select = wrapper.find('select');
+  select.element.value = 'some_index_quality';
+  select.trigger('change');
+
+  expect(refine).toHaveBeenCalledTimes(1);
+  expect(refine).toHaveBeenLastCalledWith('some_index_quality');
+});
```

Comment:
You can also add a test for the `selected` attribute, because for some reason it's not printed in the snapshots.

Reply:
I don't think I use `selected`

Reply:
Yep you do on the `option`.

https://github.com/algolia/vue-instantsearch/blob/3b4b806e0a3a903f9c64a8f3a73624ff49f57967/src/components/SortBy.vue#L16-L22